src/App.tsx

```diff
@@ -51,9 +68,45 @@ function generateHue() {
   return Math.floor(Math.random() * 360);
 }
 
+/** 
+ * This appears to still be the best way to download a file while suggesting a filename.
+ * 
+ * According to [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) 
+ * the download attribute gets ignored on URIs that are not either `same-origin` or use the `blob` or `data` schemes.
+ */
+function downloadUri(uri: string, filename: string) {
+  const downloadAnchor = document.createElement("a");
+  console.log(filename);
+  downloadAnchor.download = filename;
+  downloadAnchor.href = uri;
+  downloadAnchor.click();
+}
```

Comment:
Thanks for the docstring above this function! Could you remove `console.log()` statements though?